src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -1033,5 +1033,654 @@ public void PatternExpressionPrecedence07()
+                // switch (e) { case (
+                Diagnostic(ErrorCode.ERR_FeatureIsExperimental, "(").WithArguments("recursive patterns", "patterns2").WithLocation(1, 19),
+                // (1,20): error CS1026: ) expected
+                // switch (e) { case (
+                Diagnostic(ErrorCode.ERR_CloseParenExpected, "").WithLocation(1, 20),
+                // (1,20): error CS1003: Syntax error, ':' expected
+                // switch (e) { case (
+                Diagnostic(ErrorCode.ERR_SyntaxError, "").WithArguments(":", "").WithLocation(1, 20),
+                // (1,20): error CS1513: } expected
+                // switch (e) { case (
+                Diagnostic(ErrorCode.ERR_RbraceExpected, "").WithLocation(1, 20)
+            );
+        }
+
+        [Fact]
+        public void ParenthesizedExpression_07()
+        {
+            UsingStatement(@"switch (e) { case (): }",
+                // (1,19): error CS8058: Feature 'recursive patterns' is experimental and unsupported; use '/features:patterns2' to enable.
+                // switch (e) { case (): }
+                Diagnostic(ErrorCode.ERR_FeatureIsExperimental, "()").WithArguments("recursive patterns", "patterns2").WithLocation(1, 19)
+                );
+            N(SyntaxKind.SwitchStatement);
+            {
+                N(SyntaxKind.SwitchKeyword);
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "e");
+                }
+                N(SyntaxKind.CloseParenToken);
+                N(SyntaxKind.OpenBraceToken);
+                N(SyntaxKind.SwitchSection);
+                {
+                    N(SyntaxKind.CasePatternSwitchLabel);
+                    {
+                        N(SyntaxKind.CaseKeyword);
+                        N(SyntaxKind.DeconstructionPattern);
+                        {
+                            N(SyntaxKind.OpenParenToken);
+                            N(SyntaxKind.CloseParenToken);
+                        }
+                        N(SyntaxKind.ColonToken);
+                    }
+                }
+                N(SyntaxKind.CloseBraceToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void BrokenPattern_08()
+        {
+            UsingStatement(@"switch (e) { case",
+            // (1,18): error CS1733: Expected expression
+            // switch (e) { case
+            Diagnostic(ErrorCode.ERR_ExpressionExpected, "").WithLocation(1, 18),
+                // (1,18): error CS1003: Syntax error, ':' expected
+                // switch (e) { case
+                Diagnostic(ErrorCode.ERR_SyntaxError, "").WithArguments(":", "").WithLocation(1, 18),
+                // (1,18): error CS1513: } expected
+                // switch (e) { case
+                Diagnostic(ErrorCode.ERR_RbraceExpected, "").WithLocation(1, 18)
+                );
+        }
+
+        //[Fact]
```

Comment:
>//[Fact] [](start = 8, length = 8)

Please add a PROTOTYPE comment or remove commented out code.

Reply:
This is already tracked in the issue [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222)

---
In reply to: [151818885](https://github.com/dotnet/roslyn/pull/23208#discussion_r151818885) [](ancestors = 151818885)